src/profile_manager/views.py

```diff
@@ -103,6 +104,27 @@ def get_queryset(self):
         return self.queryset_append(profiles_qs)
 
 
+@method_decorator(staff_member_required, name='dispatch')
+class ProfileListBlock(ProfileList):
+    """lists all students in a given block, is accessed through the block list view and acts as a hybrid profile list and block detail view"""
+    view_type = ViewTypes.BLOCK
+
+    def get_queryset(self):
+        """block object is queried via pk kwarg in request from block list view, then a queryset of profiles is generated via relatedmanager"""
+        block_pk = self.kwargs['pk']
+        block = Block.objects.filter(pk=block_pk).first()
+        # queryset specifications: profile objects that are: part of active semester, a part of a coursestudent object that's in the desired block
+        profiles_qs = Profile.objects.all_for_active_semester().filter(user__coursestudent__block=block)
+        return self.queryset_append(profiles_qs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        # block object is queried again to pull name + description data from
+        context['block_object'] = Block.objects.filter(pk=self.kwargs['pk']).first()
```

Comment:
Don't add another query.  It's unnecessary (and unnecessary queries are our biggest source of lag and processor drain).

Instead, save this into a variable when you first get it in get_queryset() (add a new variable at the top of the class where you set the view_type) and then just use that variable to inject the block into the context.

Reply:
second query eschewed via implementation of block_object as a self variable.